### ISSUE TITLE
Support for request batching via /batch endpoint?

### ISSUE THREAD

**Issue by nitzmahone** [Azure/msrestazure-for-python#74]
After seeing how the portal batch-fetches `instanceView`s for PowerState so quickly via the `/batch` endpoint, I hacked together a batched version of `instance_view()` on `ComputeManagementClient`. I couldn't find any docs about the batch endpoint though (complicated by the prevalence of docs for the Batch service)..

Is `/batch` actually supported, and if so, any chance for "official" support for batched operations via msrestazure and/or the Python SDK? The bulk instanceView fetch is the only thing I can think of right now that I'd need it for, but there are probably plenty of other things out there too...

cc @lmazuel

**Comment by lmazuel** [comment]
That's super interesting, I didn't know about this one either.
@devigned @johanste ? thoughts?

**Comment by lmazuel** [comment]
https://jj09.net/azure-resource-manager-batch-api/
https://msdn.microsoft.com/en-us/skype/ucwa/batchingrequests

**Comment by lmazuel** [comment]
![image](https://user-images.githubusercontent.com/1050156/34619333-a49df532-f1f6-11e7-926e-b32b63e791e4.png)

**Comment by johanste** [comment]
I don't know why the API is not publicly documented (I have asked in the past but not gotten an authoritative answer). I may be that the endpoints don't have the SLAs needed for them to be supported for arbitrary usage. @ravbhatnagar, any insights?

**Comment by devigned** [comment]
:100: Uhm... That's something we should have documented.

**Comment by nitzmahone** [comment]
Yeah, it's definitely handy... The ARM implementation switches from a single synchronous response to a standard polled long-running operation at 20+ items (which that C# client didn't seem to take into account at first glance), and maxes out at 500 embedded requests. I don't have enough VMs laying around in my test account to poke at the limits for the instanceView use case that I care most about, but fetching instanceView for 10 VMs usually comes back in < 1s, which is a HUGE improvement over 10 additional round trips. 

I'm sure there are a number of other things this could be useful for, but the fact that it remains undocumented on ARM is definitely of concern. I'm leaning toward using it anyway for our new inventory plugin; perhaps with a switch or auto-fallback to the standard method on any unexpected failures.

**Comment by jtuliani** [comment]
Any update on getting the /batch API documented? I'd like to understand it better, for example can it do more than just GET requests, how to I specify a request body, etc.

**Comment by clarkd** [comment]
Is there any update here? I can't find any documentation for the /batch endpoint. It's also used by Azure Monitor to fetch data across multiple resources which would otherwise be many API calls.
